plugins/treasuredata/access_key.go

```diff
@@ -0,0 +1,68 @@
+package treasuredata
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func AccessKey() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.AccessKey,
+		DocsURL:       sdk.URL("https://docs.treasuredata.com/display/public/PD/TD+Toolbelt"),
+		ManagementURL: sdk.URL("https://console.treasuredata.com/app/mp/ak"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.User,
+				MarkdownDescription: "User name specified by email",
+			},
+			{
+				Name:                fieldname.APIKey,
+				MarkdownDescription: "APIKey used to authenticate to Treasure Data.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+		Importer: importer.TryAll(
```

Comment:
On a second glance, looks that there are multiple ways to provision the API Key:

```
API key specified by td -k option, which allows you to specify an API key instead of reading the config file

TREASURE_DATA_API_KEY environment variable

TD_API_KEY environment variable

TD_API_KEY in config file which path is specified by "td -c" option

TD_API_KEY in config file which path is specified by TREASURE_DATA_CONFIG_PATH environment variable

TD_API_KEY in config file which path is specified by TD_CONFIG_PATH environment variable

TD_API_KEY in config file locates at ~/.td/td.conf
```

This would mean the following:
* plugin should opt out of authentication when `-k` or `-c` flags are provided
* plugin should also try to import from `TREASURE_DATA_API_KEY` envvar
* plugin should also import from files found under `TREASURE_DATA_CONFIG_PATH` and `TD_CONFIG_PATH`

I think that the functionality can very easily be adapted to support all these use-cases, let me know if I'm missing anything! 😄

Reply:
> plugin should opt out of authentication when -k or -c flags are provided

`td` toolbelt automatically prioritizes the key given in the options (e.g. `-c` or `-k`) so I think we do not need to opt-out the authentication by the plugin by itself.

> plugin should also try to import from TREASURE_DATA_API_KEY envvar
> plugin should also import from files found under TREASURE_DATA_CONFIG_PATH and TD_CONFIG_PATH

That makes sense. I'll try to support this case too.

Reply:
> td toolbelt automatically prioritizes the key given in the options (e.g. -c or -k) so I think we do not need to opt-out the authentication by the plugin by itself.

Even if it prioritises the key given as flags, the 1Password CLI would still prompt for credentials saved in 1Password. This leads to a degraded user experience - user shouldn't have to pick a credential from 1P if they already provided one as a flag.

Reply:
@hculea Okay. But I'm not sure how to opt-out of the authentication if `-c` or `-k` options are provided. Could you give me a pointer to the resource or guide to do so?

Reply:
@hculea Okay. But I'm not sure how to opt-out of the authentication if `-c` or `-k` options are provided. Could you give me a pointer to the resource or guide to do so?

Reply:
Of course! This would be tackled in the `schema.Executable` struct, specifically within the `NeedsAuth` field. Have a look at how e.g. the `fastly` plugin approaches that:
```
		NeedsAuth: needsauth.IfAll(
			needsauth.NotForHelpOrVersion(),
			needsauth.NotWithoutArgs(),
			needsauth.NotWhenContainsArgs("-t"),
			needsauth.NotWhenContainsArgs("--token"),
			needsauth.NotWhenContainsArgs("profile"),
			needsauth.NotWhenContainsArgs("config"),
		),
```

Let me know if this makes sense!

Reply:
Great. That's what I needed. Thanks!

Reply:
Awesome! Let me know when to have another look 😄

Reply:
I have updated the patch to skip the auth if given `-c` or `-k` option.

https://github.com/1Password/shell-plugins/pull/176/commits/a1be37e6a32b0e2c9d616d33d6c0eda888efd953

Reply:
Thank you!